### DISCUSSION TITLE
How to use this library? I don't quite understand, do you have detailed textual information.

### DISCUSSION THREAD

**Discussion by txj-mssl** [TitanNano/godot-rust-script#33] [Q&A] [answered]
How to use this library? I don't quite understand, do you have detailed textual information.

**Comment by txj-mssl** [comment]
@TitanNano

**Reply by TitanNano** [reply, marked as answer]
The library is still very much work in progress, so the README is already outdated again. You have to 

1. setup the "script runtime" like this: https://github.com/TitanNano/SimChopper/blob/0e6d5cea65a1bfea6c2a41f7ec7299f79b8402b1/native/src/lib.rs
2. define a script root module. All modules under the script root module are expected to be script files. https://github.com/TitanNano/SimChopper/blob/0e6d5cea65a1bfea6c2a41f7ec7299f79b8402b1/native/src/scripts/mod.rs
3. define a script (module). The library expects a one-to-one mapping between the file name and the struct defined inside it. I.e. `dust_particles.rs` -> `DustParticles`. https://github.com/TitanNano/SimChopper/blob/0e6d5cea65a1bfea6c2a41f7ec7299f79b8402b1/native/src/scripts/particles/dust_particles.rs

The setup of the dynamic library should be done as described by gdext: https://godot-rust.github.io/book/intro/hello-world.html#create-a-rust-crate

**Reply by TitanNano** [reply]
I refactored the README with up-to-date instructions.

**Comment by txj-mssl** [comment]
Thank you. Although I know some rust syntax, after reading your explanation, I still don't quite understand the actual operation.

**Reply by TitanNano** [reply]
If you are just starting out with Rust + Godot, I recommend making yourself familiar with [godot-rust/gdext](https://www.github.com/godot-rust/gdext) first. Read the [book](https://godot-rust.github.io/book/) and [documentation](https://godot-rust.github.io/docs/gdext/master/godot/) they have, set up your project and get used to it. You don't need this library to use rust with Godot, it is merely an addition to gdext to support some specific use cases.

**Reply by txj-mssl** [reply]
ok.

**Comment by txj-mssl** [comment]
I already tried gdext yesterday, but I still don't know how to use this library. It would be great if there is a more detailed video explanation.
